File: Practica-POO/Semana01/Ejercicio06/Invoice.py
class Invoice:
    def __init__(self,product,unitprice,amount):
        self.product = product
        self.unitprice = unitprice
        self.amount = amount

    def CalculateSubTotal(self):
        return self.unitprice*self.amount

    def CalculateIGV(self):
        return self.CalculateSubTotal()*0.18

    def CalculateTotal(self):
        return self.CalculateIGV() + self.CalculateSubTotal()
    

File: Practica-POO/Semana01/Ejercicio06/Invoice.cpp
#include <iostream>
using namespace std;

class Invoice{
    public:
    string product;
    double unitprice;
    int amount;

    Invoice(string p,double u, int a){
        product = p;
        unitprice = u;
        amount = a;
    }

    double CalculateSubTotal(){
        return unitprice*amount;
    }

    double CalculateIGV(){
        return CalculateSubTotal()*0.18;
    }

    double CalculateTotal(){
        return CalculateIGV() + CalculateSubTotal();
    }
};

int main(){
    Invoice invoice1("Mermelada",8.9,4);

    double subtotal1 = invoice1.CalculateSubTotal();
    double igv1 = invoice1.CalculateIGV();
    double total1 = invoice1.CalculateTotal();
    cout << "---SUBTOTAL---\n" <<subtotal1<< "---IGV---\n" << igv1 << "---TOTAL---\n" << total1 << endl;
    
    return 0;
}